src/ORM/FieldType/DBInt.php

```diff
@@ -40,14 +40,14 @@ public function setValue(mixed $value, null|array|ModelData $record = null, bool
         parent::setValue($value, $record, $markChanged);
         // Cast string ints if they're valid
         if (is_string($this->value) && preg_match('/^-?\d+$/', $this->value)) {
-            // Ensure we can cast to int and back without loss of precision
-            // if not, keep the original value which will fail validation later
-            $stringIntValue = (string) (int) $value;
-            if ($stringIntValue !== $value) {
-                $this->value = $value;
+            // Use filter_var to validate if the string fits in a PHP integer
+            // It will return false if it overflows i.e. is "9223372036854775808"
```

Comment:
```suggestion
            // It will return false if it overflows e.g. is "9223372036854775808"
```
Another nitpick I won't block merging for. Any string bigger than PHP_INT_MAX would do this so e.g. (meaning "for example") seems more appropriate.

Reply:
Updated